amads/melody/contour/interpolation_contour.py

```diff
@@ -0,0 +1,272 @@
+"""
+Calculates the Interpolation Contour of a melody, along with related features, as
+implemented in the FANTASTIC toolbox of Müllensiefen (2009).
+"""
+
+__author__ = "David Whyatt"
+
+import numpy as np
+
+
+class InterpolationContour:
+    """Class for calculating and analyzing interpolation contours of melodies."""
```

Comment:
It would be wise to document the citation etc in the class too so that people have it at hand.